content/en/docs/concepts/cluster-administration/coordinated-leader-election.md

```diff
@@ -0,0 +1,56 @@
+---
+reviewers:
+- jpbetz
+title: Coordinated Leader Election
+content_type: concept
+weight: 200
+---
+
+<!-- overview -->
+
+{{< feature-state feature_gate_name="CoordinatedLeaderElection" >}}
+
+Kubernetes {{< skew currentVersion >}} includes an alpha feature that allows
+{{< glossary_tooltip text="control plane" term_id="control-plane" >}} components to deterministically select a leader via _coordinated leader election.
+This is useful to satisfy Kubernetes version skew constraints during cluster upgrades.
+Currently, the only builtin selection strategy is `OldestEmulationVersion`,
+preferring the leader with the lowest emulation version, followed by binary
+version, followed by creation timestamp.
+
+## Enabling coordinated leader election
+
+Ensure that `CoordinatedLeaderElection` [feature
+gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled
+when you start the {{< glossary_tooltip text="API Server"
+term_id="kube-apiserver" >}}: and that the `coordination.k8s.io/v1alpha1` API is
+enabled.
+
+This can be done by setting flags `--feature-gates="CoordinatedLeaderElection=true"` and
+`--runtime-config="coordination.k8s.io/v1alpha1=true"`.
+
+## Component Configuration
+
+With Coordinated Leader Election, components need to both run a LeaseCandidate
+and Lease goroutine (both found in client-go/pkg/leaderelection). Two components
+(kube-controller-manager and kube-scheduler) will automatically use coordinated
+leader election if enabled. Please refer to the example found in
+[k8s.io/kubernetes/cmd/kube-scheduler/app/server.go](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-scheduler/app/server.go) on set up.
+
+The created LeaseCandidate object looks similar to below:
+
+```
+apiVersion: coordination.k8s.io/v1alpha1
+kind: LeaseCandidate
+metadata:
+  name: hostname_uuid
+  namespace: kube-system
+spec:
+  binaryVersion: 1.31.0
+  emulationVersion: 1.31.0
+  leaseName: kube-scheduler
+  preferredStrategies:
+  - OldestEmulationVersion
+  renewTime: "2024-07-30T03:45:18.325483Z"
+```
```

Comment:
```suggestion
Provided that you have enabled the `CoordinatedLeaderElection` feature gate _and_
have the `coordination.k8s.io/v1alpha1` API group enabled, compatible control plane
components automatically use the LeaseCandidate and Lease APIs to elect a leader
as needed.

For Kubernetes {{< skew currentVersion >}}, two control plane components
(kube-controller-manager and kube-scheduler) automatically use coordinated
leader election when the feature gate and API group are enabled.
```

(see https://github.com/kubernetes/website/pull/46982#discussion_r1697286771 - we shouldn't be directing users to read source code).

---

If you'd like to write about this on our [contributor blog](https://k8s.dev/blog/), please do - it's a nice feature, but the whole idea is to make life simpler for end users so they don't need to care about how it works.

Reply:
ack, definitely interested in doing a write up around the more technical details without having it all littered in the documentation for end users.

Reply:
https://github.com/kubernetes/contributor-site is the place to submit the blog PR, BTW